openverse_catalog/templates/create_provider_ingester.py

```diff
@@ -0,0 +1,180 @@
+    template_path: Path,
+    provider: str,
+    endpoint: str,
+    media_types: list[str],
+    name: str,
+):
+    with target.open("w", encoding="utf-8") as target_file:
+        filled_template = _get_filled_template(
+            template_path, provider, endpoint, media_types
+        )
+        target_file.write(filled_template)
+        print(f"{name + ':':<18} {target.relative_to(PROJECT_PATH)}")
+
+
+def fill_template(provider, endpoint, media_types):
+    print(f"Creating files in {REPO_PATH}")
+
+    dags_path = TEMPLATES_PATH.parent / "dags" / "providers"
+    api_path = dags_path / "provider_api_scripts"
+    filename = inflection.underscore(provider)
+
+    # Render the API file itself
+    script_template_path = TEMPLATES_PATH / "template_provider.py_template"
+    api_script_path = api_path / f"{filename}.py"
+    _render_file(
+        api_script_path,
+        script_template_path,
+        provider,
+        endpoint,
+        media_types,
+        "API script",
+    )
+
+    # Render the tests
+    script_template_path = TEMPLATES_PATH / "template_test.py_template"
+    tests_path = PROJECT_PATH / "tests"
+    # Mirror the directory structure, but under the "tests" top level directory
+    test_script_path = tests_path.joinpath(*api_path.parts[-3:]) / f"test_{filename}.py"
+
+    _render_file(
+        test_script_path,
+        script_template_path,
+        provider,
+        endpoint,
+        media_types,
+        "API script test",
+    )
+
+    print(
+        """
+NOTE: You will also need to add a new ProviderWorkflow dataclass configuration to the \
+PROVIDER_WORKFLOWS list in `openverse-catalog/dags/providers/provider_workflows.py`.
+"""
+    )
+
+
+def sanitize_provider(provider: str) -> str:
+    """
+    Takes a provider string from user input and sanitizes it by:
+    - removing trailing whitespace
+    - replacing spaces and periods with underscores
+    - removing all characters other than alphanumeric characters, dashes,
+    and underscores.
+
+    Eg: sanitize_provider("hello world.foo*/bar2&") -> "hello_world_foobar2"
+    """
+    provider = provider.strip().replace(" ", "_").replace(".", "_")
+
+    # Remove unsupported characters
+    return re.sub("[^0-9a-xA-Z-_]+", "", provider)
+
+
+def parse_media_types(media_types: list[str]) -> list[str]:
+    """
+    Parses valid media types out from user input. Defaults to ["image",]
+    """
+    valid_media_types = []
+
+    if media_types is None:
+        media_types = []
+
+    for media_type in media_types:
+        if media_type in MEDIA_TYPES:
+            valid_media_types.append(media_type)
+        else:
+            print(f"Ignoring invalid type {media_type}")
+
+    # Default to image if no valid types given
+    if not valid_media_types:
+        print('No media type given, defaulting to ["image",]')
+        return [
+            "image",
+        ]
+
+    return valid_media_types
+
+
+def main():
+    parser = argparse.ArgumentParser(
```

Comment:
Just a note, definitely doesn't need a change: Airflow comes with [Click](https://click.palletsprojects.com/en/8.1.x/), so we could use that here for these cases if this becomes more cumbersome 🙂